### PR TITLE
[Core VM] optmize the vm

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This pr optimize the VM performance by around 5%.

Fixes # None

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo.VM/Script.cs [hunk 1]
```diff
@@ -143,11 +141,12 @@ public Script(ReadOnlyMemory<byte> script, bool strictMode)
         /// <param name="ip">The position to get the <see cref="Instruction"/>.</param>
         /// <returns>The <see cref="Instruction"/> at the specified position.</returns>
         /// <exception cref="ArgumentException">In strict mode, the <see cref="Instruction"/> was not found at the specified position.</exception>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public Instruction GetInstruction(int ip)
         {
-            if (ip >= Length) throw new ArgumentOutOfRangeException(nameof(ip));
             if (!_instructions.TryGetValue(ip, out Instruction? instruction))
             {
+                if (ip >= Length) throw new ArgumentOutOfRangeException(nameof(ip));
```
Comment (shargon) [inline src/Neo.VM/Script.cs:149]: This one is good, the other aggressive seems for big source, are you sure it improve it?

Comment (Jim8y) [reply src/Neo.VM/Script.cs:145]: they are, but for POC benchs, you are right,,, wont be that much if for normal case~~~

File: src/Neo.VM/ExecutionEngine.cs [hunk 2]
```diff
@@ -123,6 +123,7 @@ public virtual VMState Execute()
         /// <summary>
         /// Execute the next instruction.
         /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (shargon) [inline src/Neo.VM/ExecutionEngine.cs:126]:
```suggestion
```

File: src/Neo.VM/EvaluationStack.cs [hunk 3]
```diff
@@ -144,6 +144,7 @@ public T Pop<T>() where T : StackItem
             return Remove<T>(0);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (shargon) [inline src/Neo.VM/EvaluationStack.cs:147]:
```suggestion
```

**Review by shargon — COMMENTED** [review]
Can we merge it without these thwo?

**Review by shargon — APPROVED** [review]

**Pull request merged**